doc/api/node_report.md

```diff
@@ -0,0 +1,116 @@
+# node-report
+
+Delivers a human-readable diagnostic summary, written to file
+or retrieved as a text.
+
+The report is intended for development, test and production
+use, to capture and preserve information for problem determination.
+It includes JavaScript and native stack traces, heap statistics,
+platform information and resource usage etc. With the report enabled,
+reports can be triggered on unhandled exceptions, fatal errors, signals.
+
+Many capabilities are available as APIs too, for being consumed in-process.
+
+Iin-built node-report function is supported in Node.js versions 11.0.0 onwards.
+For Node.js versions 8 and below, please use it's [npm counterpart][] instead.
+
+## Usage
+
+```bash
+node --report-events fatalerror,signal,exception app.js
+```
+A report will be triggered automatically on unhandled exceptions and fatal
+error events (for example out of memory errors), and can also be triggered
+by sending a USR2 signal to a Node.js process (not supported on Windows).
+
+A report can also be triggered via an API call from a JavaScript
+application.
+
+```js
+const util = require('util');
+util.triggerReport();
+```
+The content of a report can also be returned as a JavaScript string via an
+API call from a JavaScript application.
+
+```js
+const util = require('util');
+const report_str = util.getReport();
```

Comment:
I would prefer this to go into `@nodejs/report` instead as there is some consensus in that direction.
Otherwise, I would go for `util.report.generate()` or something similar.

Reply:
Another possibility would be `util.generateDiagnosticReport()`. I'm not convinced that a new standalone module (even scoped) is necessary for this.

Reply:
My overall concern is that this adds 3 different methods to `util`. Moreover, the docs for this is separate. Even if we do not want a separate method, just having a `util.report` namespace would be more user friendly.

(I would also avoid to load this until `util.report`  is accessed).

Reply:
I'd be good with `util.report` with the methods:

* `util.report.getDiagnosticReport([error])`
* `util.report.generateDiagnosticReport([filename][, error])`
* `util.report.setDiagnosticReportOptions(options)`

Reply:
@jasnell Can we use some abstractions (classes?) instead of going with `aMethodWith20Letters`?